src/components/history-page/history-toolbar/toolbar-context/history-toolbar-state-context-provider.tsx

```diff
@@ -0,0 +1,37 @@
+/*
+ * SPDX-FileCopyrightText: 2021 The HedgeDoc developers (see AUTHORS file)
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+
+import React, { createContext, useState } from 'react'
+import type { HistoryToolbarStateWithDispatcher } from './toolbar-context'
+import { SortModeEnum } from '../../sort-button/sort-button'
+import { ViewStateEnum } from '../history-toolbar'
+import { useSingleStringUrlParameter } from '../../../../hooks/common/use-single-string-url-parameter'
+import { useArrayStringUrlParameter } from '../../../../hooks/common/use-array-string-url-parameter'
+import type { HistoryToolbarState } from '../history-toolbar-state'
+
+export const historyToolbarStateContext = createContext<HistoryToolbarStateWithDispatcher | undefined>(undefined)
+
+/**
+ * Provides a {@link React.Context react context} for the current state of the toolbar.
+ *
+ * @param children The children that should receive the toolbar state via context.
+ */
+export const HistoryToolbarStateContextProvider: React.FC = ({ children }) => {
+  const urlParameterSearch = useSingleStringUrlParameter('search', '')
+  const urlParameterSelectedTags = useArrayStringUrlParameter('selectedTags')
+
+  const stateWithDispatcher = useState<HistoryToolbarState>(() => ({
+    viewState: ViewStateEnum.CARD,
+    search: urlParameterSearch,
+    selectedTags: urlParameterSelectedTags,
+    titleSortDirection: SortModeEnum.no,
+    lastVisitedSortDirection: SortModeEnum.down
+  }))
+
+  return (
+    <historyToolbarStateContext.Provider value={stateWithDispatcher}>{children}</historyToolbarStateContext.Provider>
```

Comment:
Maybe this should be capitalized?

Reply:
No, because the var `historyToolbarStateContext` is not capitalized and I don't want to do that.

Reply:
Okay. This looks really strange this way, but if you insist that we keep it this way.